### ISSUE TITLE
Protection warrior can cap absorb with Ignore Pain

### ISSUE THREAD

**Issue by ultijlam** [Hemario/Ovale_TankScripts#2]
The protection warrior script should not recommend Ignore Pain if we are close to the the absorb cap for Ignore Pain.

**Comment by Hemario** [comment]
@ultijlam do you know what the cap is? I was looking for it in the icy veins guide but I didn't find it.

**Comment by ultijlam** [comment]
There is information on it here:
https://www.icy-veins.com/wow/protection-warrior-pve-tank-easy-mode#sec-6

You can also see how someone else implemented the caps in this WA2 tracker:
https://wago.io/IgnorePainTracker

**Comment by ultijlam** [comment]
Don't worry about implementing it if you don't have a warrior to test it on. I opened this ticket as a reminder to myself once my warrior is high enough to learn Ignore Pain. I can send a pull-up request for this.

**Comment by Hemario** [comment]
I have all tank characters, so no issue there (in fact all classes at least 110 besides a warlock).

I was looking into it today, but I found no way to obtain the exact same value as the tooltip says.
I used this formula:
`(Strength() + WeaponDPS() * 6) * 3.5 * (1 + MasteryEffect() / 100) * (1 + Versatility() / 100)`
but it came nowhere near the tooltip value.

Discovered a bug with WeaponDPS in the meantime as well.

I'm just thinking of parsing the tooltip (as the WA does) to read the exact amount, but that means we don't use the state engine.

**Comment by ultijlam** [comment]
A fellow tank enthusiast, great!

I think I'll hunt on the forums to figure out the discrepancy between the formula and the tooltip values. I'd rather get it right.

**Comment by Hemario** [comment]
Yes, I would also just get it right as well and preferably by using the state engine. It may also be possible that Ovale has its implementations of MasteryEffect() and Versatility() wrong. I don't think they are used much anymore and are still a remnant from pre-MoP days.

Thinking about the actual implementation, what would you suggest? Never propose IP until we can get the full benefit of the spell? or have a max of 30% wasted? I'd also suggest a refresh of IP in case it risks falling of.

**Comment by ultijlam** [comment]
I change my mind... I think you're right -- just grab the absorb value off of the tooltip for now.  The PaperDoll module doesn't really do state right now anyway, so we're not really benefiting from using the formula over the tooltip value.

As for clipping the absorb, I think reasonably that wasting about 25% is a good rule of thumb for now.  We can adjust that for raid content after more data and experience.
